google/cloud/bigquery/job/query.py

```diff
@@ -856,6 +856,30 @@ def configuration(self) -> QueryJobConfig:
         """The configuration for this query job."""
         return typing.cast(QueryJobConfig, super().configuration)
 
+    @property
+    def created(self):
+        """Datetime at which the job was created."""
+        millis = _helpers._int_or_none(self._job_statistics().get("creationTime"))
+        if millis is not None:
+            return _helpers._datetime_from_microseconds(millis * 1000)
+        return super().created
+
+    @property
+    def started(self):
+        """Datetime at which the job was started."""
+        millis = _helpers._int_or_none(self._job_statistics().get("startTime"))
+        if millis is not None:
+            return _helpers._datetime_from_microseconds(millis * 1000)
+        return super().started
+
+    @property
+    def ended(self):
+        """Datetime at which the job finished."""
+        millis = _helpers._int_or_none(self._job_statistics().get("endTime"))
+        if millis is not None:
+            return _helpers._datetime_from_microseconds(millis * 1000)
+        return super().ended
+
```

Comment:
I'm pretty sure these are already defined in the base class.

See: https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.job.QueryJob#google_cloud_bigquery_job_QueryJob_created

and

https://github.com/googleapis/python-bigquery/blob/3deff1d963980800e8b79fa3aaf5b712d4fd5062/google/cloud/bigquery/job/base.py#L587-L623

```suggestion
```